### PR TITLE
fix: added _ensembleValue in itemTemplateData in evaluate stage

### PR DESCRIPTION
**Pull request by sagardspeed2**
## Issue ticket number and link

Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] I have added tests
- [ ] I have added a changeset `pnpm changeset add`
- [ ] I have added example usage in the kitchen sink app

### CODE REVIEW

File: packages/runtime/src/runtime/hooks/useNavigateModal.tsx [hunk 1]
```diff
@@ -56,7 +56,7 @@ export const useNavigateModalScreen: EnsembleActionHook<
       const inputs =
         !isString(action) && action.inputs ? cloneDeep(action.inputs) : {};
       if (screenContext) {
-        const expressionMap: string[][] = [];
+        const expressionMap: [string, string, object][] = [];
```
Comment (evshi) [inline packages/runtime/src/runtime/hooks/useNavigateModal.tsx:59]: Why is this needed?

**Review by evshi — CHANGES_REQUESTED** [review]

File: packages/runtime/src/runtime/hooks/useNavigateModal.tsx (hunk 1 above)
Comment (sagardspeed2) [reply packages/runtime/src/runtime/hooks/useNavigateModal.tsx:59]: @evshi previously is was the array of string only, and I need to push the object in that array whenever it find the expression

Comment (evshi) [reply packages/runtime/src/runtime/hooks/useNavigateModal.tsx:59]: Why do you need the object...?

Comment (sagardspeed2) [reply packages/runtime/src/runtime/hooks/useNavigateModal.tsx:59]: You mean, the types is wrong or I can't use object whenever it found the expression ?
